internal/internal_activity.go

```diff
@@ -267,16 +267,19 @@ func validateFunctionAndGetResults(f interface{}, values []reflect.Value, dataCo
 			fnName, resultSize)
 	}
 
-	var result []byte
-	var err error
+	var result *commonpb.Payload
 
 	// Parse result
 	if resultSize > 1 {
 		retValue := values[0]
-		if retValue.Kind() != reflect.Ptr || !retValue.IsNil() {
-			result, err = encodeArg(dataConverter, retValue.Interface())
-			if err != nil {
-				return nil, err
+
+		var ok bool
+		if result, ok = retValue.Interface().(*commonpb.Payload); !ok {
```

Comment:
I had to add this check because previously bytes just went though `encodeArg` and now it ends up to be double encoded.